sonic-net/sonic-snmpagent
# Aggregate L3 errors with L2 discards for interfaces with RIF

Fixes https://github.com/sonic-net/sonic-buildimage/issues/19715

**- What I did**

Today we sum L2 and L3 errors on RX/TX in ifInErrors/ifOutErrors counters that are part of RFC1213 MIB.
It appears that the monitoring system is using ifInErrors as a way to detect CRC errors on physical interfaces.

The RFC vaguely defines ifInErrors as:

```
 ifInErrors OBJECT-TYPE
              SYNTAX  Counter
              ACCESS  read-only
              STATUS  mandatory
              DESCRIPTION
                      "The number of inbound packets that contained
                      errors preventing them from being deliverable to a
                      higher-layer protocol."
```

instead, aggregate errors on L3 as part of ifInDiscard and ifOutDiscard.
  

**- How I did it**

Removed aggregation logic and updated UT.

**- How to verify it**

Manual snmpwalk on interfaces - trigger L3 error - e.g. packet with dst map != router mac, check counter by snmpwalk.

Generate RIF RX errors:

```
admin@sonic:~$ show interface counters rif
    IFACE    RX_OK    RX_BPS    RX_PPS    RX_ERR    TX_OK    TX_BPS    TX_PPS    TX_ERR
---------  -------  --------  --------  --------  -------  --------  --------  --------
Ethernet0        0  0.00 B/s    0.00/s       100        0  0.00 B/s    0.00/s         0
  Vlan100        1  0.00 B/s    0.00/s         0        0  0.00 B/s    0.00/s         0
```

Before this change (14 - error, 13 - discard):
```
admin@sonic:~$ docker exec -it snmp snmpwalk -v2c -c public localhost 1.3.6.1.2.1.2.2.1.14.1
iso.3.6.1.2.1.2.2.1.14.1 = Counter32: 100
admin@sonic:~$ docker exec -it snmp snmpwalk -v2c -c public localhost 1.3.6.1.2.1.2.2.1.13.1
iso.3.6.1.2.1.2.2.1.13.1 = Counter32: 0
```

After this change:
```
admin@sonic:~$ docker exec -it snmp snmpwalk -v2c -c public localhost 1.3.6.1.2.1.2.2.1.14.1
iso.3.6.1.2.1.2.2.1.14.1 = Counter32: 0
admin@sonic:~$ docker exec -it snmp snmpwalk -v2c -c public localhost 1.3.6.1.2.1.2.2.1.13.1
iso.3.6.1.2.1.2.2.1.13.1 = Counter32: 100
```

**- Description for the changelog**
